userguide/snippets/pathlengths.rst

```diff
@@ -34,11 +34,27 @@ can also reduce the length of these limits by one or more bytes.
    | Names               |                |                                                                        |
    +---------------------+----------------+------------------------------------------------------------------------+
    | Mounted Filesystem  | 88 bytes       | Mounted filesystem path length (*MNAMELEN*). Longer paths can prevent  |
-   | Paths               |                | a device from being mounted.                                           |
+   | Paths               |                | a device from being mounted.  See below for examples of path length    |
+   |                     |                | calculation.                                                           |
    +---------------------+----------------+------------------------------------------------------------------------+
    | Device Filesystem   | 63 bytes       | `devfs(8)                                                              |
    | Paths               |                | <https://www.freebsd.org/cgi/man.cgi?query=devfs&sektion=8>`__ device  |
    |                     |                | path lengths (*SPECNAMELEN*). Longer paths can prevent a device from   |
    |                     |                | being created.                                                         |
    +---------------------+----------------+------------------------------------------------------------------------+
 
+Care is needed with regard to ZFS snapshots. **If the mounted path length for a snapshot exceeds 88 characters, 
```

Comment:
The second sentence is unclear.  A snapshot is not accessible if the mountpoint path length is more than 88 characters.  The sentence implies that you can shorten that path and the snapshot will become accessible, but what is really requiring is unmounting the snapshot, creating a new mountpoint with a shorter length, and using that.

Reply:
Also, way too many "its".  These are used correctly here, but we are trying to avoid all possessives, because they are usually used incorrectly or misunderstood by a great deal of post-literate society.

Reply:
On the need to unmounts/rename/remount - AFAIK ZFS does this automatically when you rename the dataset or components of its mounted path. You don't need to explicitly / manually unmounts it, rename it, and remount it AFAIK.  (At least, I've never had to) I have edited the para to be clearer in the case that the mountpoint itself is to be renamed.

I'm not sure that inability to handle object -> referent is a great reassurance for anyone hoping to use FreeBSD without a calamity somewhere down the line ;-) But I shall try. 3 uses of "it" and all 3 are removed. Reads more clumsy to me but can't be helped.

Reply:
@miker - yes, 1 character doesn't necessarily equal one byte in unicode and other expanded alphabets. But the correct phrasing here is a technical point of FreeBSD core OS handling that I can't answer. I just copied the existing wording on the basis it was "known good". Since the issue is a struct, I suspect it's bytes rather than characters - but let's try to avoid explaining what "Bytes" are, in this context. (An instruction to "Choose a password of at least 12 bytes length" would confuse many people, this probably isn't different)

Can you ask someone at ix to check the correct answer, and figure if we can avoid the use of "bytes" somehow?

Re Warren's point on mounting: if the scenario is ZFS automounting the snapshot, then it'll be done automatically once you rename as ZFS handles any unmount/remount in the background, so renaming is all that's needed. Only if the scenario is a manually mounted snapshot does it need manual unmounting beforehand if already mounted.